mage/mage/jar_download.clj

```diff
@@ -218,7 +219,28 @@
   (+ 3000
      (if-let [branch (:branch version)]
        (rem (Math/abs (hash branch)) 1000) ; Use hash of branch name for port
-       (major-version version)))) ; Use major version for port
+       (major-version version))))                           ; Use major version for port
+
+(defn- parse-lein-env [env-file]
+  (let [content (slurp env-file)
+        ;; Remove lines starting with ; (comments)
+        cleaned (->> (clojure.string/split-lines content)
+                     (remove #(re-matches #"^\s*;.*" %))
+                     (clojure.string/join "\n"))
+        edn-map (edn/read-string cleaned)]
+    (into {}
+          (map (fn [[k v]]
+                 [(-> k name (clojure.string/upper-case) (clojure.string/replace "-" "_")) v])
+               edn-map))))
```

Comment:
I think this is `update-keys`. you could def `(-> k name (clojure.string/upper-case) (clojure.string/replace "-" "_"))` as `->env-var` and do:

`(update-keys edn-map ->env-var)`